Modules/posixmodule.c

```diff
@@ -10067,7 +10067,9 @@ os_isatty_impl(PyObject *module, int fd)
 {
     int return_value;
     _Py_BEGIN_SUPPRESS_IPH
+    Py_BEGIN_ALLOW_THREADS
```

Comment:
Is it save to release GIL in the `_Py_BEGIN_SUPPRESS_IPH`/`_Py_END_SUPPRESS_IPH` block? I think it is okay, but want to confirm.

Reply:
See _Py_BEGIN_SUPPRESS_IPH macro: it calls _set_thread_local_invalid_parameter_handler(). The "invalid parameter handler" is set **per thread**.

Usually, the macro are written in the opposite order in posixmodule.c:

```
        Py_BEGIN_ALLOW_THREADS
        _Py_BEGIN_SUPPRESS_IPH
        res = _cwait(&status, pid, options);
        _Py_END_SUPPRESS_IPH
        Py_END_ALLOW_THREADS
```

First release the GIL, then change the invalid parameter handler.

Reply:
Right, I just noticed that too:
https://github.com/python/cpython/blob/bb3e0c240bc60fe08d332ff5955d54197f79751c/Modules/_io/fileio.c#L1128-L1132

I'll make the change.

Reply:
For the record, it is done the other way around here:
https://github.com/python/cpython/blob/c24896c0e3b32c8a9f614ef51366007b67d5c665/Python/fileutils.c#L1777-L1818

Reply:
Honestly, I don't think that the order matters. But I just consistency, when possible. It's fine to have a different order for this special case.